### ISSUE TITLE
Mex compilation problem

### ISSUE THREAD

**Issue by Lestunk** [kyamagu/mexopencv#223]
Hey I tried to compile the mex files.
But I get these errors I can't resolve:

```
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(32) : error C2039:
'CPU_MMX' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(33) : error C2039:
'CPU_SSE' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(34) : error C2039:
'CPU_SSE2' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(35) : error C2039:
'CPU_SSE3' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(36) : error C2039:
'CPU_SSSE3' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(37) : error C2039:
'CPU_SSE4_1' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(38) : error C2039:
'CPU_SSE4_2' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(39) : error C2039:
'CPU_POPCNT' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(40) : error C2039:
'CPU_AVX' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(41) : error C2039:
'CPU_AVX2' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(42) : error C2039:
'CPU_FMA3' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(43) : error C2039:
'CPU_AVX_512F' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(44) : error C2039:
'CPU_AVX_512BW' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(45) : error C2039:
'CPU_AVX_512CD' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(46) : error C2039:
'CPU_AVX_512DQ' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(47) : error C2039:
'CPU_AVX_512ER' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(48) : error C2039:
'CPU_AVX_512IFMA512' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(49) : error C2039:
'CPU_AVX_512PF' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(50) : error C2039:
'CPU_AVX_512VBMI' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(51) : error C2039:
'CPU_AVX_512VL' : is not a member of 'cv'
C:\OpenCV3\mexopencv-master\src\+cv\private\Utils_.cpp(52) : error C2039:
'CPU_NEON' : is not a member of 'cv'

Error in mexopencv.make (line 193)
            if ~opts.dryrun, eval(cmd); end
```

I Installed Opencv 3.1.0
I use Matlab 2015 64 bit. VS 2015 (also tried VS 2012 compiler)

I tested some functions GaussianBlur,... and they work.
But the function VideoCapture gives an error: 

```
     cap = cv.VideoCapture(0);
Undefined function or variable 'VideoCapture_'.

Error in cv.VideoCapture (line 109)
            this.id = VideoCapture_(0, 'new', filename);
```

Any thoughts of how to solve the compilation error? 

Thanks

**Comment by amroamroamro** [comment]
@Lestunk
Right, there were a couple of changes between 3.0.0 and 3.1.0

I've already started working on this, and I managed to fix most of the issues (the one you reported is easy to fix by simply dropping the `cv::` prefix from the enum values). I'll create a pull-request once I finish the rest...

@kyamagu
Do you want to create a `v3.1` branch similar to before, or should we just commit to master?

Anyway the current `v3.0` branch is a bit behind, and should probably be fast-forwad-merged with master before committing 3.1 stuff.

**Comment by kyamagu** [comment]
@amroamroamro Thanks for the v3.1 patch work. You can pull-request v3.1 updates directly to the master branch. I just made [a release point for v3.0.0](https://github.com/Itseez/opencv/releases/tag/3.1.0). We can direct those wishing to use v3.0 to that release.

The branch v3.0 is basically just a mirror of master and not really necessary now. Perhaps only useful when OpenCV branches the v3.x API in the future as done for v2.4.x.

**Comment by Lestunk** [comment]
Thanks for the fast reply and your work.

**Comment by Tariq-Abuhashim** [comment]
I managed to compile mexopencv using the current version of opencv (3.1.0) and opencv_contrib by not compiling the file "Utils_.cpp" in "mexopencv/src/+cv/private".
Seems that this file is the only sources of this error.

**Comment by Tariq-Abuhashim** [comment]
I managed to compile mexopencv with opencv 3.1.0 and its associated opencv_contrib my not compiling the file "Utils_.cpp" in "mexopencv/src/+cv/private". Seems that his file is the only source of error.

@Lestunk The reason why cv.VideoCapture did not work was because "VideoCapture_.cpp" is complied after "Utils_.cpp". Since it was not compiled, hence the command cv.VideoCapture was not found.

**Comment by amroamroamro** [comment]
@Lestunk, @Tariq-Abuhashim
this is fixed now that 3.1.0 updates were merged.
Pull the latest `master` branch and recompile.
